label_studio_converter/imports/yolo.py

```diff
@@ -34,40 +54,68 @@ def convert_yolo_to_ls(
     :param out_type: annotation type - "annotations" or "predictions"
     :param image_root_url: root URL path where images will be hosted, e.g.: http://example.com/images
     :param image_ext: image extension/s - single string or comma separated list to search, eg. .jpeg or .jpg, .png and so on.
+    :param yolo_type: label type - "rectanglelabels" or "polygonlabels"
     :param image_dims: image dimensions - optional tuple of integers specifying the image width and height of *all* images in the dataset. Defaults to opening the image to determine it's width and height, which is slower. This should only be used in the special case where you dataset has uniform image dimesions.
     """
 
     tasks = []
+    logger.info(f'Preparing your {out_type} yolo dataset with {yolo_type} to import into LabelStudio')
     logger.info('Reading YOLO notes and categories from %s', input_dir)
 
     # build categories=>labels dict
     notes_file = os.path.join(input_dir, 'classes.txt')
     with open(notes_file) as f:
         lines = [line.strip() for line in f.readlines()]
     categories = {i: line for i, line in enumerate(lines)}
-    logger.info(f'Found {len(categories)} categories')
+    logger.info(f'Found {len(categories)} categories:')
+    _= [logger.info(f"\t{i}: {cat}") for i, cat in enumerate(categories.values())]
+
 
     # generate and save labeling config
     label_config_file = out_file.replace('.json', '') + '.label_config.xml'
+    poly_ops = {'stroke':'3', 'pointSize':'small', 'opacity':'0.2'}
     generate_label_config(
         categories,
-        {from_name: 'RectangleLabels'},
+        {from_name: 'RectangleLabels' if yolo_type == "rectanglelabels" else 'PolygonLabels','poly_ops':poly_ops},
         to_name,
         from_name,
         label_config_file,
     )
 
     # define directories
-    labels_dir = os.path.join(input_dir, 'labels')
-    images_dir = os.path.join(input_dir, 'images')
-    logger.info('Converting labels from %s', labels_dir)
+    # retrieve data (image and label paths). handles datasets with data in subdirectories, e.g. train / val / test
+    images, labels, image_labels = get_data(input_dir, image_ext)
+    logger.info('Converting labels found recursively at %s', input_dir)
+    if yolo_type == 'polygonlabels':
+        # verify if current labels are boxes
+        # scan labels list for first non-empty label, peek contents, determine label type
+        for label in labels:
+            with open(labels[0]) as f:
+                sample_lbl = [line.strip() for line in f.readlines()]
+            if len(sample_lbl) == 0:
+                continue
+            else:
+                break   # non-empty label found
+        logger.info(f'sample label: {sample_lbl}')
+        if len(sample_lbl) < 7: # Polygons expected to consist of 7 items. At least three x,y pairs + class 
```

Comment:
I could be wrong here, but I believe that `sample_lbl` will contain a list of strings, each one being the line of the label file (where lines are separate objects). To test whether the sample label has less than 7 fields you should test something like `if len(sample_lbl[0].split(' ')) < 7`